File: Dot examples/ASP Book Examples/Book Examples/Chapter15/VB/App_Code/Employees.vb

Public MustInherit Class BaseEmployee

    Public MustOverride ReadOnly Property Salary() As Decimal

    Public ReadOnly Property Company() As String
        Get
            Return "Acme Software"
        End Get
    End Property

End Class

Public Class SalesEmployee
    Inherits BaseEmployee

    Public Overrides ReadOnly Property Salary() As Decimal
        Get
            Return 67000.23D
        End Get
    End Property

End Class

File: Dot examples/ASP Book Examples/Book Examples/Chapter15/CS/App_Code/Employees.cs
public abstract class BaseEmployee
{
    public abstract decimal Salary
    {
        get;
    }
   

    public string Company
    {
        get { return "Acme Software"; }
    }
}

public class SalesEmployee : BaseEmployee
{
    public override decimal Salary
    {
        get { return 67000.23m; }
    }

}